src/Components/Components/src/Forms/ValidationSummary.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.RenderTree;
+using System;
+
+namespace Microsoft.AspNetCore.Components.Forms
```

Comment:
Do these control-likes belong in the `.Browser` assembly since these are coupled to HTML? I guess NavLink is in this assembly. Not urgent if we do want to change it.

Reply:
I agree 100%, and in fact the same thing occurred to me over the weekend, and I was planning to do it.

However after trying it, I think it's better if we don't do so right now. The problem is that you then become forced to add `@addTagHelper *, Microsoft.AspNetCore.Components.Browser` in all the projects and Razor component libraries where you want to use those components. We can't auto-add those tag helpers in the default compilation, because that breaks the whole point of the layering. And `@addTagHelper` is extremely non-discoverable in the case where you don't even know where a given component comes from (how do you know to add `Microsoft.AspNetCore.Components.Browser` specifically?).

So, once we replace `@addTagHelper` with `@using`, I think we should move the `Input*` and `NavLink` components to `.Browser`.

Reply:
Alright great. I totally agree and hadn't thought about it in that level of detail yet, but I like your plan.